app/Domain/Integrations/Events/ActivationExpired.php

```diff
@@ -11,7 +11,7 @@ final class ActivationExpired
 {
     use Dispatchable;
 
-    public function __construct(public readonly UuidInterface $id)
+    public function __construct(public readonly UuidInterface $id, public readonly string $templateName)
```

Comment:
Maybe make the full class readonly?

Is there a reason to pass the string value instead of the enum type here?

Reply:
The difference between an enum and a string is not that big. There is also a feature in Laravel that supports this, see https://laravel.com/docs/11.x/queues#class-structure